### ISSUE TITLE
Imgui unable to handle Inputs except Keyboard Nav.

### ISSUE THREAD

**Issue by sravankaruturi** [ocornut/imgui#2542]
**Version/Branch of Dear ImGui:**

Version: 1.71
Branch: 1.71 _master_

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_win32.cpp + imgui_impl_dx11.cpp 
Compiler: msvc v142 _(if the question is related to building or platform specific features)_
Operating System: Windows 10

**My Issue/Question:**

ImGUI not handling any inputs apart from the Keyboard Navigation Inputs.

**Screenshots/Video**

![ImGUIInputQuestion](https://user-images.githubusercontent.com/10460714/57311278-50e5aa00-70b9-11e9-9716-f533c2a4e2f1.gif)

**Standalone, minimal, complete and verifiable example:** 

https://github.com/sravankaruturi/Platformer

Code when Initializing ImGui:

	IMGUI_CHECKVERSION();
	ImGui::CreateContext();

	ImGuiIO& io = ImGui::GetIO(); (void)io;
	io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;

	ImGui::StyleColorsDark();

	ImGui_ImplWin32_Init(hWnd);
	ImGui_ImplDX11_Init(device, context);

	io = ImGui::GetIO();
	io.DisplaySize.x = width;
	io.DisplaySize.y = height;

Code to handle Inputs:

```
    extern LRESULT ImGui_ImplWin32_WndProcHandler(HWND hWnd, UINT msg, WPARAM wParam, LPARAM lParam);
    LRESULT DXCore::WindowProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
    {

	if (ImGui_ImplWin32_WndProcHandler(hWnd, uMsg, wParam, lParam)) {
		return true;
	}

	return DXCoreInstance->ProcessMessage(hWnd, uMsg, wParam, lParam);
    }

```

```

**Comment by ocornut** [comment]
Hello,

Your Win32 message processing is probably incorrect. Please look into using a debugger and printf logging to figure out what's wrong with your application code, if you can confirm you are indeed sending mouse data maybe trace in imgui_impl_win32.cpp.

(You are not providing a "Standalone, minimal, complete and verifiable example:" merely a whole application.)

**Comment by ocornut** [comment]
You can browse in `Inputs,Navigation & Focus -> Keyboard, Mouse & Navigation State` to see what dear imgui is receiving from the back-end.

If you can confirm that the examples/ application are working on your machine, the difference is probably due to something your code is doing.

One fault I could see in imgui_impl_win32.cpp is that recent version are testing for the `GetForegroundWindow` in `ImGui_ImplWin32_UpdateMousePos()`. You may try to comment out those two lines:

```
    if (HWND active_window = ::GetForegroundWindow())
        if (active_window == g_hWnd || ::IsChild(active_window, g_hWnd))
```

And see if it works without.

**Comment by sravankaruturi** [comment]
The same code worked for the Docking branch. However, I had a few glitches and decided to use the master branch where I started facing this issue.

Every time I hover over an Imgui window, I can see the breakpoint triggered in the function, ImGui_ImplWin32_WndProcHandler()

I tried commenting out the two lines as you mentioned in the ImGui_ImplWin32_UpdateMousePos() function, but that didn't help.

**Comment by sravankaruturi** [comment]
Also, that application is quite barebones. I'm sorry, but could you elaborate a bit on Standalone, minimal, complete and verifiable example. Thanks again for responding.

**Comment by ocornut** [comment]
See if the demo applications in the examples/ folder work.
If they do and yours doesn't, I suppose you'll have to find what's different in yours.

**Comment by ocornut** [comment]
It looks like you are not calling `ImGui_ImplWin32_NewFrame()` at all, that's your bug.

**Comment by sravankaruturi** [comment]
Oh. So sorry. I must have missed when the Intellisense completed it. Thanks so much for talking the time.
